src/main/java/com/hierynomus/smbj/share/NamedPipe.java

```diff
@@ -111,7 +120,48 @@ public int read(byte[] buffer, int offset, int length) {
      * @return the output message
      */
     public byte[] transact(byte[] inBuffer) {
-        return ioctl(FSCTL_PIPE_TRANSCEIVE, true, inBuffer, 0, inBuffer.length);
+        final SMB2IoctlResponse ioctlResponse = share.ioctl(
+            fileId,
+            FSCTL_PIPE_TRANSCEIVE,
+            true,
+            new ArrayByteChunkProvider(inBuffer, 0, inBuffer.length, 0),
+            -1,
+            SUCCESS_OR_BUFFER_OVERFLOW
+        );
+
+        final ByteArrayOutputStream outBuffer = new ByteArrayOutputStream(4096);
+        final byte[] ioctlOutputBuffer = ioctlResponse.getOutputBuffer();
+        try {
+            outBuffer.write(ioctlOutputBuffer);
+        } catch (final IOException exception) {
+            throw new SMBRuntimeException(exception);
+        }
+
+        final NtStatus ioctlStatus = ioctlResponse.getHeader().getStatus();
+        if (ioctlStatus.equals(NtStatus.STATUS_BUFFER_OVERFLOW)) {
+            for (;;) {
```

Comment:
I don't think it's necessary to move this to another method.  This code is reading the pipe after the transaction failed; so the first thing it does is add the transaction response buffer to the out buffer which is the first part of the named pipe response.  It follows that up with a loop that continues to read until no more data is pending from the named pipe.  In this scenario, all the data is consumed, and we make the least number of calls possible (we don't remake the transaction call) so this implementation is the most efficient solution.  However, because the out buffer is not bounded, a malicious server could in theory cause the client to run out of memory.

I can move this to a separate method, perhaps "readFully", if you want.  Might even make sense in the share itself to be re-used for other situations where the data needs to be fully read.